### ISSUE TITLE
Possible feature: action icon next to search bar

### ISSUE THREAD

**Issue by EclipticWld** [umhan35/react-native-search-bar#144]
I've noticed that the library doesn't have the option to set showing "+" button while not editing. It would be cool to this one with ability to set a color for the icon.

![](https://media.giphy.com/media/3ohc12sqIE1SpUweTm/giphy.gif)

**Comment by EclipticWld** [comment]
@iRoachie what do you think about it?

**Comment by iRoachie** [comment]
Seems like a very specific use case. Can you not have a container with the searchbar and your icon button next to it?

**Comment by EclipticWld** [comment]
This is Reminders inbuild iOS App. So it's hard to say that it's a very specific use case. I can make a container but it takes more space on the screen to place [Search][Cancel][+]

**Comment by iRoachie** [comment]
Would the `cancelButtonText` on the `next` branch work for your use case?

**Comment by EclipticWld** [comment]
I don't think so. I'm not forcing to implement features for my needs. I'm just offering ways to extend functionality and if it's along with my needs it's cool.

Just to clarify Remiders app states:
```
initial state: [Search][+]
onFocus: [Search][Cancel]
onTextExist: [Search][x][Cancel]
```

If we add manually [+] button it would be

```
initial state: [Search][+]
onFocus: [Search][Cancel][+]
onTextExist: [Search][x][Cancel]
```

If pass `showsCancelButtonWhileEditing` it would be

```
initial state: [Search][+]
onFocus: [Search][+]
onTextExist: [Search][x][+]
```
